components/ShoppingItem/ShoppingItem.js

```diff
@@ -3,49 +3,63 @@ import Link from "next/link";
 import { useState } from "react";
 
 export default function ShoppingItem({
-  id,
   onDeleteItem,
   shoppingItem,
   onEditItem,
   onChangeMode,
-  mode,
+  onToggleIsPurchased,
+  purchasedItems,
 }) {
   const [isToBeDeleted, setIsToBeDeleted] = useState(false);
 
-  // const [isToBeEdited, setIsToBeEdited] = useState(false);
-
   function toggleIsToBeDeleted() {
     setIsToBeDeleted(!isToBeDeleted);
   }
 
-  // function toggleIsToBeEdited() {
-  //   setIsToBeEdited(!isToBeEdited);
-  // }
-
   return (
     <ListItem
-      key={shoppingItem.id}
       $backgroundColor={shoppingItem.backgroundColor}
+      $isPurchased={shoppingItem.isPurchased}
     >
       <CategoryBox>{shoppingItem.category}</CategoryBox>
-
       {!isToBeDeleted ? (
         <>
-          {shoppingItem.name}: {shoppingItem.quantity}
+          <StyledTickboxNameNumberContainer>
+            <form>
+              <input
+                type="checkbox"
+                id={`checkbox-${shoppingItem.id}`}
+                checked={shoppingItem.isPurchased}
+                onChange={() => {
+                  onToggleIsPurchased(shoppingItem.id);
+                }}
+              />
+              <label htmlFor={`checkbox-${shoppingItem.id}`}>
+                {purchasedItems.includes(shoppingItem.id) ? "purchased" : ""}
+              </label>
+            </form>
+            <span
+              style={
+                shoppingItem.isPurchased
+                  ? { textDecoration: "line-through", color: "#404040" }
+                  : undefined
+              }
+            >
+              {shoppingItem.name}: {shoppingItem.quantity}
+            </span>
+          </StyledTickboxNameNumberContainer>
           <EditButton
             onClick={() => {
               onChangeMode();
               onEditItem();
```

Comment:
See my comment in your `index.js` file. What I suggested is basically exactly what you are doing here, just one level above (in the `index.js`), so that you do not need to pass down the `onChangeMode` as a prop.

Anyhow, you are missing the parameter for the `onChangeMode` function here. You wanna set the mode to "edit".

Reply:
not sure what you are referring to; maybe we could meet up and talk about this.

Reply:
I understand that you finally want to finish this user story. If you want me to explain my suggestion, we can meet up - no problem. 🙂 Anyhow, this is just a minor change.

Reply:
Hey Andrea, 
it would be great if you could explain this to me because I don't understand what I should do.